### PR TITLE
fix(search): Search by substring implemented

### PR DESCRIPTION
**Pull request by AlexandrZagorskiy**

### CODE REVIEW

File: components/scg/kb/sources/controls/ui_control_search.scs [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ui_control_search
 	<- ui_scg_control; sc_node_not_relation;
 	=> nrel_main_idtf:
-		[Системный идентификатор...]
+		[идентификатор...]
 		(* <- lang_ru;; *);
 	=> nrel_main_idtf:
-        [System identifier...]
+        [identifier...]
```
Comment (MikhailSadovsky) [inline components/scg/kb/sources/controls/ui_control_search.scs:7]: Pleas update to "Search"

File: components/scg/kb/sources/controls/ui_control_search.scs [hunk 2]
```diff
@@ -1,8 +1,8 @@
 ui_control_search
 	<- ui_scg_control; sc_node_not_relation;
 	=> nrel_main_idtf:
-		[Системный идентификатор...]
+		[идентификатор...]
```
Comment (MikhailSadovsky) [inline components/scg/kb/sources/controls/ui_control_search.scs:4]: Please update to "Поиск"

File: scripts/install_deps_ubuntu.sh [hunk 3]
```diff
@@ -1,2 +1,2 @@
 sudo apt-get install -y python3-pip
-sudo yes | pip3 install tornado==4.2 sqlalchemy redis==2.9 numpy configparser
+sudo yes | pip3 install tornado==4.2 sqlalchemy rocksdb==0.7.0 numpy configparser
```
Comment (MikhailSadovsky) [inline scripts/install_deps_ubuntu.sh:2]: "progress" module should be added as well

File: server/handlers/api.py [hunk 4]
```diff
@@ -305,83 +302,57 @@ def post(self):
...
-                        array.insert(idx, data)
-                        inserted = True
-                        break;
-                    idx = idx + 1
-                
-                if not inserted and len(array) < max_n:
+            if data not in array:
+                if len(array) > 0:
+                    idx = 0
+                    max_n = 100
```
Comment (MikhailSadovsky) [inline server/handlers/api.py:327]: why it was updated to 100 instead of tornado option?

File: server/handlers/api_logic.py [hunk 5]
```diff
@@ -259,6 +259,26 @@ def find_translation_with_format(construction_addr, format_addr, keynode_nrel_fo
     return None
 
 
+@decorators.method_logging
+def get_by_system_identifier(keys, sctp_client, idtf):
+    keynode_nrel_system_identifier = keys[KeynodeSysIdentifiers.nrel_system_identifier]
+    links = sctp_client.find_links_with_content(idtf)
+    if links:
+        for l in links:
```
Comment (MikhailSadovsky) [inline server/handlers/api_logic.py:267]: please rename "l" to "link"

File: server/handlers/api_logic.py [hunk 6]
```diff
@@ -310,29 +330,31 @@ def get_identifier_translated(addr, used_lang, keys, sctp_client):
...
-                                                     SctpIteratorType.SCTP_ITERATOR_3F_A_F,
-                                                     used_lang,
-                                                     ScElementType.sc_type_arc_pos_const_perm,
-                                                     l
-                                                     )
-                if langs is not None:
-                    return elements[0][0]
-
-    return sctp_client.find_element_by_system_identifier(idtf)
+            for k in keynode_idtfs:
```
Comment (MikhailSadovsky) [inline server/handlers/api_logic.py:340]: please rename "k" to "keynode_idtf"

File: server/reader_rocksdb.py [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    addr_postfix = "_addrs"
+    long_content = None
+
+    def __init__(self):
+        self.sys = []
+        self.main = []
+        self.common = []
+        self.addr = []
+        self.long_content_counter = 0
+        self.content_len_max = 200
```
Comment (MikhailSadovsky) [inline server/reader_rocksdb.py:24]: Could you please describe 200 count?

Comment (AlexandrZagorskiy) [reply server/reader_rocksdb.py:24]: > Could you please describe 200 count?
Links with 100+ symbols have a high chance to drop exception while decoding it. 200 is an optimal value between performance and number of items to search, loss of items less than 1 percent.

**Review by MikhailSadovsky — APPROVED** [review]

**Pull request merged**